### ISSUE TITLE
Generating k-mer detection networks for large datasets

### ISSUE THREAD

**Issue by hasindu2008** [cvdelannoy/baseLess#3]
Will it be possible to use nanopolish/f5c eventalign instead of Tombo to generate k-mer detection networks for large references/datasets? Tombo does not scale well with large datasets unfortunately.

**Comment by cvdelannoy** [comment]
That's a great proposal! Haven't tried, doesn't sound too tough if it does the same thing, though if the hdf5 path or output format of f5c is different from tombo's it wouldn't work out of the box. I'll take a look as soon as I have a moment to see if I can easily add support.

**Comment by hasindu2008** [comment]
Its output is not HDF5, but a simple table-separated text format. So it should be much easier to parse than accessing HDF5. Do you use tombo to get: 1) the signal to basecalled read alignment or 2) the signal to reference genome alignment?
Both options are available in f5c and I can tell you where exactly to look if you provide a bit more information like what information in the tombo output you are using?

**Comment by cvdelannoy** [comment]
Thanks, help is always welcome! I'd prefer signal to reference genome alignment to take out some erroneous base calls. What we use is the base to raw measurement index table returned by tombo, though if we immediately can assign events to k-mers that would also be fine. I figure I'll need `f5c eventalign --signal-index` , right? I'll be looking at this in two weeks I think.

**Comment by hasindu2008** [comment]
You are right. You could use the `--print-read-names` option so that read IDs are printed instead of the read index. The simplest set of commands would be (I will explain you the performance-tuned options later):
```
minimap2 -a -x map-ont genome.fa reads.fastq | samtools sort -o reads.sorted.bam
samtools index reads.sorted.bam
f5c index -d fast5_files/ reads.fastq 
f5c eventalign -b reads.sorted.bam -g genome.fa -r reads.fastq --signal-index  --print-read-names > events.tsv
```
```
contig	position	reference_kmer	read_name	strand	event_index	event_level_mean	event_stdv	event_length	model_kmer	model_mean	model_stdv	standardized_level	start_idx	end_idx
chr1	8338535	GCACCG	9f2bfb40-ed0b-4d00-bf21-fc682efc23bf	t	6219	91.69	0.954	0.00075	CGGTGC	92.04	2.58	-0.12	31612	31615
chr1	8338536	CACCGG	9f2bfb40-ed0b-4d00-bf21-fc682efc23bf	t	6218	99.54	1.708	0.00125	CCGGTG	101.06	2.98	-0.46	31607	31612
chr1	8338536	CACCGG	9f2bfb40-ed0b-4d00-bf21-fc682efc23bf	t	6217	102.96	2.724	0.00100	CCGGTG	101.06	2.98	0.58	31603	31607
chr1	8338537	ACCGGG	9f2bfb40-ed0b-4d00-bf21-fc682efc23bf	t	6216	123.51	0.378	0.00100	CCCGGT	122.31	2.12	0.51	31599	31603
chr1	8338537	ACCGGG	9f2bfb40-ed0b-4d00-bf21-fc682efc23bf	t	6215	121.61	0.832	0.00175	CCCGGT	122.31	2.12	-0.30	31592	31599
chr1	8338538	CCGGGC	9f2bfb40-ed0b-4d00-bf21-fc682efc23bf	t	6214	125.03	0.946	0.00225	GCCCGG	125.06	1.74	-0.02	31583	31592
chr1	8338538	CCGGGC	9f2bfb40-ed0b-4d00-bf21-fc682efc23bf	t	6213	124.51	1.093	0.00125	GCCCGG	125.06	1.74	-0.29	31578	31583
chr1	8338538	CCGGGC	9f2bfb40-ed0b-4d00-bf21-fc682efc23bf	t	6212	125.65	0.811	0.00125	GCCCGG	125.06	1.74	0.31	31573	31578
chr1	8338538	CCGGGC	9f2bfb40-ed0b-4d00-bf21-fc682efc23bf	t	6211	124.09	1.399	0.00200	GCCCGG	125.06	1.74	-0.51	31565	31573
```

I think the columns you would require are only:
```read_name, start_idx, end_idx```
Perhaps contig and position to deduce the base to raw signal mapping.

Currently, for training, do you grab the raw signal for each k-mer? Is it the k-mers from the reference you use? And what is the k-mer size? I can provide a custom output option if anything needed is missing.

In the Tombo output after you get the base to the raw measurement index table, do you revisit the raw signal to grab the raw samples?
